File: Selenium/Activities/Activity10.py
from selenium import webdriver
from selenium.webdriver.chrome.service import Service as chromeService
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver import ActionChains
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
service=chromeService(ChromeDriverManager().install())
with webdriver.Chrome(service=service) as driver:
    actions = webdriver.ActionChains(driver)
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    pageTitle=driver.title
    print(pageTitle)
    wait=WebDriverWait(driver,20)
    checkbox=driver.find_element(By.CLASS_NAME,"willDisappear");
    removeCheckbox=driver.find_element(By.ID,"toggleCheckbox");
    print("Checkbox is visible",checkbox.is_displayed)
    removeCheckbox.click()
    wait=wait.until(EC.invisibility_of_element(driver.find_element(By.CLASS_NAME,"willDisappear")))
    print("Checkbox is visible",checkbox.is_displayed)
    removeCheckbox.click()
    print("Checkbox is visible",checkbox.is_displayed)
    driver.find_element(By.XPATH,"//input[@class='willDisappear']").click()
    driver.close()

File: Selenium/Activities/Activity10.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity10 {
    public static void main(String[] args) {
        WebDriverManager.chromedriver().setup();
        WebDriver driver = new ChromeDriver();
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of webpage : " + driver.getTitle());
        WebElement checkbox=driver.findElement(By.className("willDisappear"));
        WebElement removeCheckbox=driver.findElement(By.id("toggleCheckbox"));
        if(checkbox.isDisplayed()){
            System.out.println("Checkbox is visible");
        }else{
            System.out.println("Checkbox is not visible");
        }
        removeCheckbox.click();
        WebDriverWait wait=new WebDriverWait(driver, Duration.ofSeconds(60));
        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.className("willDisappear")));
        System.out.println("Checkbox is displayed: " + checkbox.isDisplayed());
        removeCheckbox.click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("dynamicCheckbox")));
        System.out.println("Checkbox is displayed: " + checkbox.isDisplayed());
        driver.findElement(By.xpath("//input[@class='willDisappear']")).click();
        driver.close();

    }
}